File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        
        d = dict()
        ans = -1
        for i in s:
            if i not in d.keys():
                d[i] = 1
            else:
                d[i] += 1
   
        for i in d.keys():
            if d[i] == 1:
                ans = s.index(i)
                break
        return ans
        

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cs
public class Solution {
    public int FirstUniqChar(string s) {
        var c = s.GroupBy(chr => chr)
                 .Select(g => new {k = g.Key, v = g.Count()})
                 .Where(e => e.v == 1)
                 .FirstOrDefault()?.k??' ';
        if(c != ' ')
        {
            return s.IndexOf(c);
        }
        return -1;
    }
}